app/src/main/java/io/heckel/ntfy/work/DeleteWorker.kt

```diff
@@ -59,6 +63,27 @@ class DeleteWorker(ctx: Context, params: WorkerParameters) : CoroutineWorker(ctx
         }
     }
 
+    private fun deleteExpiredIcons() {
+        Log.d(TAG, "Deleting icons for deleted notifications")
+        val resolver = applicationContext.contentResolver
+        val repository = Repository.getInstance(applicationContext)
+        val activeIconUris = repository.getActiveIconUris()
+        val expiredIconUris = repository.getDeletedIconUris()
+        val urisToDelete = expiredIconUris.minus(activeIconUris)
+        urisToDelete.forEach { uri ->
+            try {
+                val deleted = resolver.delete(Uri.parse(uri), null, null) > 0
+                if (!deleted) {
+                    Log.w(TAG, "Unable to delete icon at $uri")
+                }
+
+                repository.clearIconUri(uri)
+            } catch (e: Exception) {
+                Log.w(TAG, "Failed to delete icon: ${e.message}", e)
+            }
+        }
+    }
```

Comment:
This looks great, except that there is still a chance of leftover icons, right? Say you delete a subscription, then you'll delete all notifications and all references of icons. Ideally you'd do:

```
val activeIconUris = repository.getActiveIconUris()
val onDiskIconUris = filesystem.listAllIconUris()
val urisToDelete = onDiskIconUris.minus(activeIconUris)
```

I wonder if this is also a problem for attachments. :thinking: